File: create_vercel_kv.py
import json
import os

def generate_puppeteer_commands():
    """
    Generates and prints a series of mcp__puppeteer JSON commands
    to automate the creation of a Vercel KV database.
    """
    # --- 1. 读取凭证 (Read Credentials) ---
    # 构建凭证文件的绝对路径
    # The script must read credentials from ~/.claude/secrets/vercel_credentials.json
    credential_path = os.path.expanduser('~/.claude/secrets/vercel_credentials.json')

    try:
        with open(credential_path, 'r') as f:
            credentials = json.load(f)
        email = credentials.get('email')
        password = credentials.get('password')
        if not email or not password:
            print(json.dumps({
                "error": "Email or password not found in credentials file.",
                "path": credential_path
            }))
            return
    except FileNotFoundError:
        print(json.dumps({
            "error": "Credentials file not found.",
            "path": credential_path
        }))
        return
    except json.JSONDecodeError:
        print(json.dumps({
            "error": "Could not decode JSON from credentials file.",
            "path": credential_path
        }))
        return

    # 定义团队和项目信息
    # Team slug and project name are provided in the prompt.
    team_slug = "addisons-team-u4znpcv"
    project_name = "gemini-balance-lite"
    storage_url = f"https://vercel.com/{team_slug}/{project_name}/storage"
    login_url = "https://vercel.com/login"

    # --- 2. 生成自动化指令 (Generate Automation Commands) ---
    # 这是一个指令列表，将按顺序执行
    commands = [
        # 步骤 1: 导航至登录页面
        # Step 1: Navigate to the login page
        {
            "tool_name": "mcp__puppeteer__navigate",
            "parameters": {
                "url": login_url,
                "options": {
                    "waitUntil": "networkidle2"
                }
            },
            "comment": "Navigate to Vercel login page."
        },
        # 步骤 2: 填充邮箱并继续
        # Step 2: Fill in the email and continue
        {
            "tool_name": "mcp__puppeteer__fill",
            "parameters": {
                "selector": "input[name='email']",
                "text": email
            },
            "comment": "Fill in the email address."
        },
        {
            "tool_name": "mcp__puppeteer__click",
            "parameters": {
                "selector": "button[type='submit']"
            },
            "comment": "Click the 'Continue with Email' button."
        },
        # 步骤 3: 填充密码并登录
        # Step 3: Fill in the password and log in
        {
            "tool_name": "mcp__puppeteer__fill",
            "parameters": {
                "selector": "input[name='password']",
                "text": password
            },
            "comment": "Fill in the password. A wait is added for the password field to appear."
        },
        {
            "tool_name": "mcp__puppeteer__click",
            "parameters": {
                "selector": "button[type='submit']"
            },
            "comment": "Click the final login button."
        },
        # 步骤 4: 导航至项目的 Storage 页面
        # Step 4: Navigate to the project's Storage page
        {
            "tool_name": "mcp__puppeteer__navigate",
            "parameters": {
                "url": storage_url,
                "options": {
                    "waitUntil": "networkidle2"
                }
            },
            "comment": "Navigate directly to the project's storage page after login."
        },
        # 步骤 5: 创建新的 KV 数据库
        # Step 5: Create the new KV database
        {
            "tool_name": "mcp__puppeteer__click",
            "parameters": {
                # This selector targets a button that specifically creates a KV store.
                "selector": "button[aria-label='Create a new KV store']"
            },
            "comment": "Click the 'Create Database -> KV' button."
        },
        # 步骤 6: 确认创建流程
        # Step 6: Go through the confirmation process
        {
            "tool_name": "mcp__puppeteer__click",
            "parameters": {
                # The 'Continue' button in the creation dialog.
                "selector": "button:contains('Continue')"
            },
            "comment": "Click 'Continue' in the creation dialog."
        },
        {
            "tool_name": "mcp__puppeteer__click",
            "parameters": {
                # The final 'Accept' or 'Create' button.
                "selector": "button:contains('Accept & Create')"
            },
            "comment": "Accept the terms and create the database."
        },
        # 步骤 7: 截取成功页面作为验证
        # Step 7: Take a screenshot as visual confirmation
        {
            "tool_name": "mcp__puppeteer__screenshot",
            "parameters": {
                "path": "/tmp/vercel_kv_creation_success.png",
                "fullPage": True
            },
            "comment": "Take a screenshot to verify successful creation."
        }
    ]

    # --- 3. 打印指令 (Print Commands) ---
    # 将每个指令字典转换为格式化的 JSON 字符串并打印
    for command in commands:
        # Adding a comment to the JSON output for clarity.
        print(f"# {command.pop('comment')}")
        print(json.dumps(command, indent=2))
        print("-" * 20)


if __name__ == "__main__":
    generate_puppeteer_commands()

File: create_vercel_kv.mjs

import puppeteer from 'puppeteer';

const VERCEL_PROJECT_URL = 'https://vercel.com/addisons-team/gemini-balance-lite';
const KV_DATABASE_NAME = 'gemini-balance-lite-kv';

async function main() {
    console.log('启动交互式浏览器自动化...');
    const browser = await puppeteer.launch({
        headless: false,
        defaultViewport: null, // Use the browser's default viewport
        args: ['--start-maximized'] // Start browser maximized
    });
    const page = await browser.newPage();

    try {
        console.log('导航到 Vercel 登录页面...');
        await page.goto('https://vercel.com/login', { waitUntil: 'networkidle2' });

        console.log('自动点击 "Continue with GitHub"...');
        const githubButton = await page.waitForSelector('button[data-testid="sso-provider-button-github"]');
        await githubButton.click();

        console.log('\n================================================================');
        console.log('=== 操作已暂停，等待您手动完成 GitHub 登录和授权... ===');
        console.log('=== 请在这个新打开的 Chrome 窗口中完成操作。           ===');
        console.log('=== 登录成功后，脚本将自动接管后续步骤。           ===');
        console.log('================================================================\n');

        await page.waitForNavigation({ timeout: 300000, waitUntil: 'networkidle2' }); // 5 minutes timeout for manual login

        // Wait until the user has logged in and is on a dashboard-like page
        await page.waitForFunction(
            () => document.querySelector('nav[aria-label="Dashboard"] a[href="/dashboard"]') !== null,
            { timeout: 300000 }
        );

        console.log('检测到您已成功登录！脚本现在将自动接管...');

        console.log('导航到项目的 "Storage" 标签页...');
        await page.goto(`${VERCEL_PROJECT_URL}/storage`, { waitUntil: 'networkidle2' });

        console.log('检查 KV 数据库是否已存在...');
        const dbExists = await page.evaluate((name) => {
            const heading = Array.from(document.querySelectorAll('h3')).find(h3 => h3.textContent.trim() === name);
            return !!heading;
        }, KV_DATABASE_NAME);


        if (dbExists) {
            console.log(`KV 数据库 "${KV_DATABASE_NAME}" 已存在。跳过创建步骤。`);
        } else {
            console.log('点击 "Create Database" 并选择 KV Store...');
             const createButton = await page.waitForSelector('button[aria-label="Create a new store"]');
            await createButton.click();

            const kvOption = await page.waitForSelector('div[role="option"] p:nth-child(1)');
            await kvOption.click();

            console.log('输入数据库名称...');
            const nameInput = await page.waitForSelector('input#create-kv-database-name-input');
            await nameInput.type(KV_DATABASE_NAME);

            console.log('提交创建表单...');
            // We need to find the submit button inside the dialog
            const dialog = await page.waitForSelector('div[role="dialog"]');
            const submitButton = await dialog.$('button[type="submit"]');
            await submitButton.click();


            console.log('等待数据库创建完成...');
            await page.waitForFunction(
                (name) => {
                    const heading = Array.from(document.querySelectorAll('h3')).find(h3 => h3.textContent.trim() === name);
                    return !!heading;
                },
                { timeout: 60000 },
                KV_DATABASE_NAME
             );


            console.log(`成功创建 KV 数据库 "${KV_DATABASE_NAME}"。`);
        }

        console.log('\n自动化流程成功完成！');

    } catch (error) {
        console.error('自动化过程中发生错误:', error);
        try {
            const screenshotPath = 'interactive_error_screenshot.png';
            await page.screenshot({ path: screenshotPath });
            console.log(`已将错误截图保存为: ${screenshotPath}`);
        } catch (screenshotError) {
            console.error('无法保存截图:', screenshotError);
        }
    } finally {
        console.log('浏览器将在10秒后自动关闭...');
        await new Promise(resolve => setTimeout(resolve, 10000));
        await browser.close();
    }
}

main();
